# find product

In [4]:
n=int(input())
x=input()
answer = 1
for i in x.split():
    answer = (answer*int(i))%(1000000000+7)
print(answer)


5
1 2 3 4 5
120


# play with numbers

In [1]:
n = input().split()
n[0], n[1] = int(n[0]),int(n[1])
a = input().split() # Read array elements
sum = [] # Initialize comulative sum array
# Comulative sum
for i in range(0, n[0]):
    if i == 0:
        sum.append(int(a[i]))
    else:
        sum.append(int(sum[i-1])+int(a[i]))
del a
# Read each query and calculate the average
for k in range(0, n[1]):
    inq = input().split()
    i = int(inq[0])
    j = int(inq[1])
    if i>1:
        print((sum[j-1]-sum[i-2])// (j-i+1))
    else:
        print(sum[j-1]//(j-i+1))



5 3
1 2 3 4 5
1 3
2
2 4
3
2 3
2


# seating arrangement

In [ ]:
t=int(input())
for i in range(1,t+1):
    seatnum=int(input())
    n=(seatnum-1)//12
    if seatnum%3==2:
        s="MS"
    elif seatnum%6<2:
        s="WS"
    else:
        s="AS"
    print(24*n+13-seatnum,s)
                    

# very easy problems

# 1.prime number

### 1.You are given an integer N. You need to print the series of all prime numbers till N.

### Input Format

- The first and only line of the input contains a single integer N denoting the number till where you need to find the series of prime number.

- Output Format

- Print the desired output in single line separated by spaces.

- Constraints

- 1<=N<=1000

- sample i/p : 9   
- sample o/p : 2,3,5,7


In [1]:
def prime(n):
    for num in range(2, n + 1):
            for i in range(2, num):
                if (num % i) == 0:
                    break
            else:
                 print(num, end=" ")
    return
num = int(input())
prime(num)

9
2 3 5 7 

# 2.Seating Arrangement

#### Akash and Vishal are quite fond of travelling. They mostly travel by railways. They were travelling in a train one day and they got interested in the seating arrangement of their compartment. The compartment looked something like So they got interested to know the seat number facing them and the seat type facing them. The seats are denoted as follows : 

- Window Seat : WS
- Middle Seat : MS
- Aisle Seat : AS

- You will be given a seat number, find out the seat number facing you and the seat type, i.e. WS, MS or AS.

- INPUT
- First line of input will consist of a single integer T denoting number of test-cases. Each test-case consists of a single integer N denoting the seat-number.

- OUTPUT
- For each test case, print the facing seat-number and the seat-type, separated by a single space in a new line.

- CONSTRAINTS
- 1<=T<=105
- 1<=N<=108

- sample i/p :- 2                      sample o/p:- 19 WS
                18                                  45 AS
                40              

In [1]:
t=int(input())
for i in range(1,t+1):
    seatnum=int(input())
    n=(seatnum-1)//12
    if seatnum%3==2:
        s="MS"
    elif seatnum%6<2:
        s="WS"
    else:
        s="AS"
    print(24*n+13-seatnum,s)
                    

2
18
19 WS
40
45 AS


# 3.Palindromic String

- You have been given a String S. You need to find and print whether this string is a palindrome or not. If yes, print "YES" (without quotes), else print "NO" (without quotes).

- Input Format
- The first and only line of input contains the String S. The String shall consist of lowercase English alphabets only.

- Output Format
- Print the required answer on a single line.

- Constraints 1 < |S| < 100

- Note
- String S consists of lowercase English Alphabets only.

- sample i/p: aba             output: yes

In [2]:
def palindrome(st):
    sl=st[::-1]
    if sl==st:
        print("YES")
    else:
        print("NO")
        
st=input()
palindrome(st)


aba
YES


# 4.Find Product

- You have been given an array A of size N consisting of positive integers. You need to find and print the product of all the number in this array Modulo 10^9+7.

- Input Format:
- The first line contains a single integer N denoting the size of the array. The next line contains N space separated integers denoting the elements of the array

- Output Format:
- Print a single integer denoting the product of all the elements of the array Modulo 10^9+7.
- input : 5                     - output :- 120
-         1 2 3 4 5       



In [3]:
n=int(input())
x=input()
answer = 1
for i in x.split():
    answer = (answer*int(i))%(1000000000+7)
print(answer)


5
1 2 3 4 5
120


# 5.Count Divisors

#### You have been given 3 integers - l, r and k. Find how many numbers between l and r (both inclusive) are divisible by k. You do not need to print these numbers, you just have to find their count.

- Input Format
- The first and only line of input contains 3 space separated integers l, r and k.

- Output Format
- Print the required answer on a single line.

- i/p:- 1 10 1             
- o/p: 10

In [4]:
def countDevisors(g, j, k): 
    count = 0
    for i in range (g, j+1):

        if i % k  == 0:
            count = count + 1
    print(count)
    return
s = input()
s = s.split()
g = int(s[0])
j = int(s[1])
k = int(s[2])
countDevisors(g, j, k)

1 10 1
10


# 6.Factorial!

#### You have been given a positive integer N. You need to find and print the Factorial of this number. The Factorial of a positive integer N refers to the product of all number in the range from 1 to N. You can read more about the factorial of a number here.

- Input Format:
- The first and only line of the input contains a single integer N denoting the number whose factorial you need to find.

- Output Format
- Output a single line denoting the factorial of the number N.

- Constraints 1<=N<=10

- i/p:- 2           
- o/p:- 2

In [6]:
def fact(n):
    fact = 1
    for i in range (1,n+1):
        fact = fact*i
    print(fact)
    return
n=int(input())
fact(n)
        

2
2


# 7.Toggle String

#### You have been given a String S consisting of uppercase and lowercase English alphabets. You need to change the case of each alphabet in this String. That is, all the uppercase letters should be converted to lowercase and all the lowercase letters should be converted to uppercase. You need to then print the resultant String to output.

- Input Format
- The first and only line of input contains the String S

- Output Format
- Print the resultant String on a single line.

- Constraints 1<= S <= 100 (where 's' denotes length of the string)

- i/p:- aBcD    
- o/p:- AbCd



In [7]:
def toggle(s):
    for i in s:
        if i == i.upper():
            n = i.lower()
        elif i == i.lower():
            n = i.upper()
        print(n, end = "")
s = input()
toggle(s)


acVGhjB
ACvgHJb

# 8.Roy and Profile Picture

#### Roy wants to change his profile picture on Facebook. Now Facebook has some restriction over the dimension of picture that we can upload.
- Minimum dimension of the picture can be L x L, where L is the length of the side of square.

- Now Roy has N photos of various dimensions.
- Dimension of a photo is denoted as W x H 
- where W - width of the photo and H - Height of the photo

- When any photo is uploaded following events may occur:

- [1] If any of the width or height is less than L, user is prompted to upload another one. Print "UPLOAD ANOTHER" in this case.
- [2] If width and height, both are large enough and 
- (a) if the photo is already square then it is accepted. Print "ACCEPTED" in this case.
- (b) else user is prompted to crop it. Print "CROP IT" in this case.

- (quotes are only for clarification)

- Given L, N, W and H as input, print appropriate text as output.

- Input:
- First line contains L.
- Second line contains N, number of photos.
- Following N lines each contains two space separated integers W and H.

- Output:
- Print appropriate text for each photo in a new line.

- Constraints:
- 1 <= L,W,H <= 10000
- 1 <= N <= 1000

- i/p:- 180
        3
        640 480
        120 300
        180 180